src/components/ThemeToggle.tsx

```diff
@@ -0,0 +1,38 @@
+import { Button } from '@/components/ui/button';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { useThemeStore } from '@/store/theme';
+import { Moon, Sun } from 'lucide-react';
+
+function ThemeToggle() {
+  const setTheme = useThemeStore((state) => state.setTheme);
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button size="icon" variant="outline">
+          <Sun className="size-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0" />
+          <Moon className="absolute size-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100" />
+          <span className="sr-only">Toggle theme</span>
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end">
+        <DropdownMenuItem onClick={() => setTheme('light')}>
```

Comment:
To można wynieść poza JSX np:

```
type ThemeTypes = "light" | "dark" | "system";

const handleUpdateTheme = (theme: ThemeTypes) => () => {
  setTheme(theme);
}
```

a w JSX przekazujesz tlyko:
`onClick={handleUpdateTheme('wartosc')}`

Reply:
Zwroc uwage, ze wykonujesz funkcje z parametrem, ktora zwraca funkcje ktora juz ten parametr zapameita :) czysciej w JSX i funkcja wyniesiona.